src/math/trig.c

```diff
@@ -74,8 +100,10 @@ void cordic_sin_cos(int32_t th_rad_fxp, cordic_cfg type, int32_t *sign, int32_t
 	xtmp = cordic_sine_cos_lut_q29fl;
 	ytmp = 0;
 
-	if ((type == EN_16B_CORDIC_SINE) || (type == EN_16B_CORDIC_COSINE))
-		a_idx = CORDIC_SIN_COS_15B_TABLE_SIZE;
+	if (type == EN_16B_CORDIC_SINE ||
```

Comment:
I feel like rather than passing in the type of function we are running and having the algorithm figure the switches out, we should pass in a configuration struct and each function should configure the algorithm itself, therefore pushing the complexity out of the core and allowing you to add new functions without modifying the core.

Reply:
ping on removing this check in core alg, this should be passed in